lib/canql/nodes/anomaly.rb

```diff
@@ -10,13 +11,51 @@ def anomaly_type
           ".#{text_value}"
         end
 
+        def anomaly_status_type
+          text_value = status_type.text_value.strip
+          return '' if '' == text_value
+
+          ".#{text_value}"
+        end
+
+        def anomaly_qualifier
+          anomaly_status_type + anomaly_type
+        end
+
         def meta_data_item
+          existance_filter.merge(code_filter)
+        end
+
+        def existance_filter
           {
-            "anomaly#{anomaly_type}.exists" => {
+            "anomaly#{anomaly_qualifier}.exists" => {
               Canql::EQUALS => existance_modifier.text_value.strip != 'no'
             }
           }
         end
+
+        def code_filter
+          return {} if code_data.text_value.blank?
+          code_array = [code_data.first.to_code]
+          code_data.rest.elements.map do |e|
```

Comment:
Can we avoid single-char variables please.

Reply:
and I think `each` would suffice here, as the `map` return value is unused?

Reply:
I'll make the changes, but you should note that this code was lifted out of the TNQL gem, so someone should probably make similar change in that project.